core/src/main/scala/org/apache/spark/storage/BlockInfoManager.scala

```diff
@@ -340,7 +340,7 @@ private[storage] class BlockInfoManager extends Logging {
     val blocksWithReleasedLocks = mutable.ArrayBuffer[BlockId]()
 
     val readLocks = synchronized {
-      readLocksByTask.remove(taskAttemptId).get
+      readLocksByTask.remove(taskAttemptId).getOrElse(ImmutableMultiset.of[BlockId]())
```

Comment:
Missed this PR earlier .. sorry about that.
Since this is an unexpected scenario (it really must not happen according to current code !), we should probably have logged the stack trace and warning message in the logs to help debug.
It silently ignores a fairly bad bug otherwise.